src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/crafting/FurnaceListener.java

```diff
@@ -0,0 +1,34 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners.crafting;
+
+import javax.annotation.Nonnull;
+
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.inventory.InventoryMoveItemEvent;
+import org.bukkit.event.inventory.InventoryType;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+
+public class FurnaceListener implements SlimefunCraftingListener {
+
+    public FurnaceListener(@Nonnull SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onFurnace(InventoryMoveItemEvent e) {
+        Location loc = e.getDestination().getLocation();
+        if (loc != null && BlockStorage.hasBlockInfo(loc) && e.getSource().getType() == InventoryType.HOPPER) {
+            boolean isElectricOreGrinder = BlockStorage.check(loc, SlimefunItems.ELECTRIC_ORE_GRINDER.getItemId()) || BlockStorage.check(loc, SlimefunItems.ELECTRIC_ORE_GRINDER_2.getItemId());
```

Comment:
Yes thjs would work but wouldn't it be really tedious to keep adding machines to this that are also furnaces i think the best way to do this is to check if its a furnace and it has an id then cancel the event

Reply:
Now that you pointed it out, yes this would be tedious to manage. The way you suggested, won't this affect enhanced, reinforced and carbonado furnaces? Also, won't this cause issue for other furnace based addon machines which wants to retain this "cooking/smelting" functionality?

Reply:
Then maybe some sort of implementation if you want it toggled on or off 
Then people can decide thereself?
Just throwing in ideas

Reply:
How about an abstract `class SlimefunContainer` extending `AContainer` and then extending the `SlimefunContainer` in all those furnace based machines which extends `AContainer` and the listener cancels the `InventoryMoveItemEvent` from hopper if the destination block is an instance of `SlimefunContainer`? 

**TL;DR:** Like Fluffy did in commit https://github.com/NCBPFluffyBear/FluffyMachines/commit/52b6fec3b64e62498f41ee15b61de5385145466b

Reply:
That could work aswell